chain/messagepool/messagepool.go

```diff
@@ -1078,14 +1078,42 @@ func (mp *MessagePool) getStateNonce(ctx context.Context, addr address.Address,
 		return n, nil
 	}
 
-	act, err := mp.api.GetActorAfter(addr, ts)
+	raddr, err := mp.resolveToKey(ctx, addr)
```

Comment:
`raddr` is only used much later in the `for` loop. It was suspicious for me that we assigned `raddr` but then kept on using `addr` in `GetActorBefore`. Wdyt about moving this line just before `for`?